tests/sentry/api/serializers/test_project.py

```diff
@@ -259,6 +259,69 @@ def test_avoid_hidden_and_no_env(self):
         assert result['latestRelease'] == {'version': self.release.version}
         assert result['environments'] == ['production', 'staging']
 
+    def test_multiple_environments_deploys(self):
+        env_1_release = self.create_release(self.project)
+        env_1_deploy = Deploy.objects.create(
+            environment_id=self.environment_1.id,
+            organization_id=self.organization.id,
+            release=env_1_release,
+            date_finished=self.date + timedelta(minutes=20),
+        )
+        ReleaseProjectEnvironment.objects.create(
+            project_id=self.project.id,
+            release_id=env_1_release.id,
+            environment_id=self.environment_1.id,
+            last_deploy_id=env_1_deploy.id,
+        )
+
+        env_2_release = self.create_release(self.project)
+        Deploy.objects.create(
+            environment_id=self.environment_2.id,
+            organization_id=self.organization.id,
+            release=env_2_release,
+            date_finished=self.date - timedelta(days=5),
+        )
+        env_2_deploy = Deploy.objects.create(
+            environment_id=self.environment_2.id,
+            organization_id=self.organization.id,
+            release=env_2_release,
+            date_finished=self.date,
+        )
```

Comment:
Should you have more than one deploy to the same release environment to test the ordering conditions?

Reply:
I think that's covered? The deploy on line 278 has the same env/release/project. We can't have duplicate release project environment rows, so we can't have multiple rows that have the same env/release/project and a different last deploy id. And the data on 270 creates a second RPE for `self.project`, which should handle the ordering there.

Let me know if I'm missing something though, the relationships are kind of convoluted.

Reply:
I misread the deploy record creation.